File: program/src/pda.rs
use solana_program::pubkey::Pubkey;

pub const PREFIX: &str = "JSON";

pub fn find_metadata_account(json_account: &Pubkey) -> (Pubkey, u8) {
    Pubkey::find_program_address(
        &[PREFIX.as_bytes(), crate::ID.as_ref(), json_account.as_ref()],
        &crate::id(),
    )
}

File: packages/sdk/src/pda.ts
import { PublicKey } from '@solana/web3.js';
import { PROGRAM_ADDRESS } from './generated';

export const PREFIX: string = 'JSON';

export function find_metadata_account(json_account: PublicKey) {
  return PublicKey.findProgramAddressSync(
    [Buffer.from(PREFIX), new PublicKey(PROGRAM_ADDRESS).toBuffer(), json_account.toBuffer()],
    new PublicKey(PROGRAM_ADDRESS),
  );
}
